presto-hive/src/main/java/io/prestosql/plugin/hive/HdfsConfig.java

```diff
@@ -67,6 +69,20 @@ public HdfsConfig setResourceConfigFiles(List<File> files)
         return this;
     }
 
+    @NotNull
+    @Pattern(regexp = "0[0-7]{3}", message = "must be octal number, with leading 0")
+    public String getNewDirectoryPermissions()
+    {
+        return newDirectoryPermissions;
+    }
+
+    @Config("hive.dfs.new-directory-permissions")
```

Comment:
Add `@ConfigDescription`

Reply:
Name this `hive.fs` since this isn’t specific to HDFS

The `dfs` names are very legacy and newer ones are named `hdfs`